### PR TITLE
vala: split into vala-doc, vala-vapi, valadoc, valadoc-devel

### PR DESCRIPTION
**Pull request by maxice8**

### CODE REVIEW

**Comment by maxice8** [comment]
vala -> main package which provides valac, vapigen and all that stuff that we need in hostmakedepends
libvala -> libvala library, used by vala and only 2 other packages (gnome-builder and io.elementary.code)
vala-devel -> libvala development files, used for linking against libvala
valadoc -> valadoc library, no packages use it
valadoc-devel -> valadoc development files, used for linking against libvaladoc
vala-doc -> devhelp documentation for vala
vala-vapi -> common vapi files used during generation (vala depends on it), but it is separated as it can be used standalone in makedepends= while cross-compiling

**Comment by Cogitri** [comment]
LGTM to me.

**Comment by maxice8** [comment]
> LGTM to me.

1. Thanks
2. TM in LGTM is 'to me'

**Comment by Anachron** [comment]
Not sure if sarcasm ... LGTM = Lets Get This Merged.

**Comment by Duncaen** [comment]
http://www.mirbsd.org/wtf.cgi?q=LGTM
```
$ netbsd-wtf lgtm
LGTM: looks good to me
```

File: common/hooks/pre-configure/02-script-wrapper.sh [hunk 1]
```diff
@@ -85,7 +85,7 @@ vapigen_wrapper() {
 	[ -x ${XBPS_WRAPPERDIR}/vapigen ] && return 0
 	cat >>${XBPS_WRAPPERDIR}/vapigen<<_EOF
 #!/bin/sh
-exec /usr/bin/vapigen \\
+exec /usr/bin/vapigen-0.42 \\
```
Comment (Vaelatern) [inline common/hooks/pre-configure/02-script-wrapper.sh]: Why versioning this?

**Comment by maxice8** [comment]
On Wed, Feb 06, 2019 at 06:56:22PM +0000, Toyam Cox wrote:
> Vaelatern commented on this pull request.
> 
> 
> 
> > @@ -85,7 +85,7 @@ vapigen_wrapper() {
>  	[ -x ${XBPS_WRAPPERDIR}/vapigen ] && return 0
>  	cat >>${XBPS_WRAPPERDIR}/vapigen<<_EOF
>  #!/bin/sh
> -exec /usr/bin/vapigen \\
> +exec /usr/bin/vapigen-0.42 \\
> 
> Why versioning this?
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/8239#pullrequestreview-200746274

No reason, leftovers from some experimentation

**Pull request merged**